### PR TITLE
REP-4575 Fix the assumption that the Request's Input Stream will support mark/reset.

### PR DESCRIPTION
**Pull request by wdschei**

### CODE REVIEW

**Comment by wdschei** [comment]
This is the Jenkins build: https://jenkins.openrepose.org/job/repose-pull_requests/1303/

**Comment by wdschei** [comment]
Retest build: https://jenkins.openrepose.org/job/repose-pull_requests/1304/

**Comment by Mario-Lopez** [comment]
Both builds failed due to time bombs since both targeted a 7.3.7.0 commit instead of your latest commit on this branch.

File: repose-aggregator/functional-tests/spock-functional-test/src/test/configs/features/filters/rackspaceauthuser/inputstream/system-model.cfg.xml [hunk 1]
```diff
@@ -0,0 +1,22 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<system-model xmlns="http://docs.openrepose.org/repose/system-model/v2.0">
+    <repose-cluster id="repose">
+
+        <nodes>
+            <node id="node1" hostname="localhost" http-port="${reposePort}"/>
+        </nodes>
+
+        <filters>
+            <filter name="${preceding-filter}"/>
```
Comment (adrianjgeorge) [inline repose-aggregator/functional-tests/spock-functional-test/src/test/configs/features/filters/rackspaceauthuser/inputstream/system-model.cfg.xml]: I saw this and thought you were going to do a super sexy unroll, now I has a sad.

Comment (Mario-Lopez) [inline repose-aggregator/functional-tests/spock-functional-test/src/test/configs/features/filters/rackspaceauthuser/inputstream/system-model.cfg.xml]: I totally was, but I couldn't figure out how to do it without starting Repose 90 times.  Eventually I chucked it in the bucket. :-(

**Pull request merged**